### PR TITLE
Rename transaction

### PR DESCRIPTION
**Pull request by cschneid**
This matches the Ruby agent's ScoutApm::Transaction.rename call.

This is most useful when an app has a single entry point, then makes business
logic decisions with easily nameable branches. GraphQL for instance has a
single POST endpoint, but immediately branches into different queries and
mutations with clear names.

**NOTE** - This requires a yet-unreleased change to the binary agent, and will
not yet work. A note will be added to this PR when it is fully released

### CODE REVIEW

**Comment by dlanderson** [comment]
@cschneid Is this supported in core agent yet?

**Comment by cschneid** [comment]
It isn't, I'll add it to the list of next version things. Created https://github.com/scoutapp/core-agent/issues/69 to track.

**Comment by adamchainz** [comment]
👍

File: tests/scout_apm_tests/api/rename_transaction_test.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import scout_apm.api
```
Comment (adamchainz) [inline tests/scout_apm_tests/api/rename_transaction_test.py]: I think the test should be in `integration/test_api.py` with all the other tests for the public intefrace

File: tests/scout_apm_tests/api/rename_transaction_test.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+from scout_apm.core.tracked_request import TrackedRequest
+
+
+def test_rename_transaction():
+    tr = TrackedRequest.instance()
+    assert('transaction_name' not in tr.tags)
```
Comment (adamchainz) [inline tests/scout_apm_tests/api/rename_transaction_test.py]: no brackets, though `black` will sort that for you, also shouldn't that be `transaction.name` with a dot?

Comment (cschneid) [reply tests/scout_apm_tests/api/rename_transaction_test.py]: yes, it should be

**Pull request merged**